src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1742,8 +1717,8 @@ fn resolution_failure(
                         for ns in [TypeNS, ValueNS, MacroNS] {
                             if let Ok(v_res) = collector.resolve(start, ns, item_id, module_id) {
                                 debug!("found partial_res={:?}", v_res);
-                                if !v_res.is_empty() {
-                                    *partial_res = Some(full_res(tcx, v_res[0]));
+                                if let Some(&res) = v_res.first() {
+                                    *partial_res = Some(full_res(tcx, res));
```

Comment:
Pattern matching instead of checking `.is_empty()` and then doing bounds-checked indexing.